packages/bff/src/services/catalogService.ts

```diff
@@ -592,5 +561,54 @@ export function catalogServiceBuilder(
           riskAnalysisId,
         },
       }),
+    exportEServiceDescriptor: async (
+      eserviceId: EServiceId,
+      descriptorId: DescriptorId,
+      { authData, headers, logger }: WithLogger<BffAppContext>
+    ): Promise<bffApi.FileResource> => {
+      const requesterId = authData.organizationId;
+
+      const eservice = await catalogProcessClient.getEServiceById({
+        params: {
+          eServiceId: eserviceId,
+        },
+        headers,
+      });
+
+      assertRequesterIsProducer(requesterId, eservice);
+
+      const zipFolderName = `${eservice.id}_${descriptorId}`;
+      const zipFile = await createDescriptorDocumentZipFile(
+        bffConfig.s3Bucket,
+        fileManager,
+        logger,
+        zipFolderName,
+        eservice,
+        descriptorId
+      );
+
+      const zipFileName = `${requesterId}_${zipFolderName}.zip`;
+      const filename = await fileManager.storeBytes(
+        {
+          bucket: bffConfig.exportEserviceContainer,
+          path: bffConfig.exportEservicePath,
+          name: zipFileName,
+          content: Buffer.from(zipFile),
+        },
+        logger
+      );
+
+      const url = await fileManager.generateGetPresignedUrl(
+        bffConfig.exportEserviceContainer,
+        bffConfig.exportEservicePath,
+        zipFileName,
+        bffConfig.presignedUrlGetDurationMinutes
+      );
+
+      return {
+        filename,
```

Comment:
```suggestion
        filename: `${zipFolderName}.zip`,
```
I think this should be like this, in order to return only the name of the file instead of the whole path on S3

Reply:
`filename` contains stored S3 object's key, that already contains the file extension (`.zip`)

Reply:
Yes but in this case the frontend is expecting only the name of the zip without the whole path; `filename` here contains the whole path of the zip as we can see in your tests

Reply:
👍🏻

Reply:
[e82f37d](https://github.com/pagopa/interop-be-monorepo/pull/882/commits/e82f37dae3b6dcbf0bab9301734887200cad42ac) should resolve the problem